configure.ac

```diff
@@ -2040,6 +2040,20 @@ if test "x$with_kvm_openfiles" = "xyes"; then
   with_libkvm="yes"
 fi
 
+AC_CHECK_HEADERS([libpfctl.h], [], [],
+[
+  #include <sys/queue.h>
+  #include <sys/types.h>
+  #include <netinet/in.h>
+  #include <net/if.h>
+  #include <net/pfvar.h>
+])
```

Comment:
This check is inconsequential: neither the success nor the failure case have any side effects.

Option 1: If you simplify this to

```
AC_CHECK_HEADERS([libpfctl.h])
```

then autoconf's default behavior will be active, which will create a `HAVE_LIBPFCTL_H` define if the header is present. You can use it instead of `__FreeBSD__` in the .c file.

Option 2: In the unlikely case that you have to specify the other headers, skipping the arguments entirely will trigger the default behavior (I think):

```
AC_CHECK_HEADERS([libpfctl.h],,, # <-- empty arguments
[
  #include <sys/queue.h>
  #include <sys/types.h>
  #include <netinet/in.h>
  #include <net/if.h>
  #include <net/pfvar.h>
])
```

Option 3: remove entirely.

Reply:
Case in point, the build currently fails with:

```
src/pf.c:39:10: fatal error: 'libpfctl.h' file not found
#include <libpfctl.h>
         ^~~~~~~~~~~~
1 error generated.
--- src/pf.lo ---
*** [src/pf.lo] Error code 1
```

This is despite the configure script technically checking for the header's existence:

```
checking for libpfctl.h... no
checking for pfctl_status_counter in -llibpfctl... no
```

Reply:
Hmm, the build works for me where I have libpfctl present:

```
checking for libpfctl.h... yes
checking for pfctl_status_counter in -lpfctl... yes
```

I had to add the other includes so that the test worked.. but if we don't need it, I am happy to remove it.

Where are you seeing that build failure? It looks like the checks on this PR are successful?

Reply:
I see some failed checks below. The detailed view of the FreeBSD one is here: https://cirrus-ci.com/task/6128106226843648

Reply:
ah, thank you!  I am not very familiar with the github actions and didn't see where I could get that detail.. the in github link didn't show any errors, just looked like it built successfully.

In the FreeBSD port, we set `CPPFLAGS= -I/usr/local/include` :
https://cgit.freebsd.org/ports/tree/net-mgmt/collectd5/Makefile#n84

Looks like that is needed in the .cirrus.yml...

Reply:
There is a typo in the flag definition (`CPPLAGS` instead of `CPPFLAGS`).

Reply:
We definitely need `HAVE_LIBPFCTL_H` and cannot rely on `__FreeBSD__`, see comment below. Please try if

```
AC_CHECK_HEADERS([libpfctl.h])
```

works, that would be the easiest.